File: HW/HW7/HW7.py
def minMeetingRooms(self, intervals: List[List[int]]) -> int:       
        start = []
        end = [] 

        for i in range(len(intervals)):
            start.append(intervals[i][0])

        for i in range(len(intervals)):
            end.append(intervals[i][1])

        start.sort()
        end.sort()

        s = 0
        e = 0
        servers = 0

        while s < len(start):
            if start[s] < end[e]:
                s += 1
                servers += 1
            else:
                e += 1
                servers -= 1

        return servers

File: HW/HW7/HW7.java
public int minMeetingRooms(int[][] intervals) { 
      int[] start = new int[intervals.length];
      int[] end = new int[intervals.length];

      for(int i = 0; i < intervals.length; i++){
          start[i] = intervals[i][0];
      }

      for(int i = 0; i < intervals.length; i++){
          end[i] = intervals[i][1];
      }
      
      
      
      int[] sortedStart = sort(start);
      int[] sortedEnd = sort(end);

      int s = 0;
      int e = 0;
      int servers = 0;

      while(s < sortedStart.length){
          if(sortedStart[s] < sortedEnd[e]){
              s++;
              servers++;
          } else {
              e++;
              servers--;
          }
      }

      return servers;
}

public int[] sort(int[] arr){
    int n = arr.length;
    int[] newArr = arr;
    for (int i = 1; i < n; ++i) {
        int key = newArr[i];
        int j = i - 1;

        while (j >= 0 && newArr[j] > key) {
            newArr[j + 1] = newArr[j];
            j = j - 1;
        }
        newArr[j + 1] = key;
    }
    return newArr;
}
